controller/setting_controller.go

```diff
@@ -1385,26 +1398,57 @@ func (info *ClusterInfo) collectSettings() error {
 
 	for name, value := range settingMap {
 		structName := util.StructName(fmt.Sprintf(ClusterInfoSettingFmt, util.ConvertToCamel(name, "-")))
-		info.structFields.Append(structName, value)
+
+		switch reflect.TypeOf(value).Kind() {
+		case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64, reflect.Float32, reflect.Float64:
+			info.structFields.fields.Append(structName, value)
+		default:
+			info.structFields.tags.Append(structName, fmt.Sprint(value))
+		}
 	}
 	return nil
 }
 
+func (info *ClusterInfo) convertSettingValueType(setting *longhorn.Setting) (convertedValue interface{}, err error) {
+	definition, ok := types.GetSettingDefinition(types.SettingName(setting.Name))
+	if !ok {
+		return false, fmt.Errorf("failed to get Setting %v definition", setting.Name)
+	}
+
+	switch definition.Type {
+	case types.SettingTypeInt:
+		return strconv.ParseInt(setting.Value, 10, 64)
+	case types.SettingTypeBool:
+		return strconv.ParseBool(setting.Value)
+	default:
+		return setting.Value, nil
+	}
+}
+
 func (info *ClusterInfo) collectVolumesInfo() error {
 	volumesRO, err := info.ds.ListVolumesRO()
 	if err != nil {
 		return errors.Wrapf(err, "failed to list Longhorn Volumes")
 	}
+	volumeCount := len(volumesRO)
+
+	isV2DataEngineEnabled, err := info.ds.GetSettingAsBool(types.SettingNameV2DataEngine)
+	if err != nil {
+		return err
+	}
 
 	var totalVolumeSize int
 	var totalVolumeActualSize int
 	var totalVolumeNumOfReplicas int
-	accessModeCountStruct := make(map[util.StructName]int)
-	dataLocalityCountStruct := make(map[util.StructName]int)
-	frontendCountStruct := make(map[util.StructName]int)
+	accessModeCountStruct := make(map[util.StructName]int, volumeCount)
+	dataLocalityCountStruct := make(map[util.StructName]int, volumeCount)
+	frontendCountStruct := make(map[util.StructName]int, volumeCount)
 	for _, volume := range volumesRO {
-		totalVolumeSize += int(volume.Spec.Size)
-		totalVolumeActualSize += int(volume.Status.ActualSize)
+		isVolumeV2DataEngineEnabled := isV2DataEngineEnabled && volume.Spec.BackendStoreDriver == longhorn.BackendStoreDriverTypeV2
+		if !isVolumeV2DataEngineEnabled {
+			totalVolumeSize += int(volume.Spec.Size)
```

Comment:
Only `actual size` is unable to supported by v2 data engine, so `volume.spec.size` still needs to include.